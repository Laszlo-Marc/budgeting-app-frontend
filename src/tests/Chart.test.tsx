/* eslint-disable @typescript-eslint/no-explicit-any */
import {render} from '@testing-library/react';
import {vi} from 'vitest';
import BasicPie from '../components/expenseComponents/Chart';
Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: vi.fn().mockImplementation((query: any) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: vi.fn(), // deprecated
        removeListener: vi.fn(), // deprecated
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
        dispatchEvent: vi.fn(),
    })),
});

describe('render Chart component corectly', () => {
    it('should render Chart component correctly', () => {
        const {container} = render(<BasicPie />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
