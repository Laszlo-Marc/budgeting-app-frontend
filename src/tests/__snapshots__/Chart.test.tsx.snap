// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render Chart component corectly > should render Chart component correctly 1`] = `
<div
  class="MuiBox-root css-i9gxme"
>
  <div
    class="css-1sfpt6l-MuiResponsiveChart-container"
  >
    <svg
      class="css-13aj3tc-MuiChartsSurface-root"
      height="400"
      viewBox="0 0 400 400"
      width="400"
    >
      <title />
      <desc />
      <g>
        <g
          transform="translate(152.5, 200)"
        >
          <g>
            <path
              class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
              cursor="unset"
              d="M0,-73.75L0,0Z"
              style="fill: red; opacity: 1;"
            />
            <path
              class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
              cursor="unset"
              d="M0,-73.75L0,0Z"
              style="fill: blue; opacity: 1;"
            />
            <path
              class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
              cursor="unset"
              d="M0,-73.75L0,0Z"
              style="fill: green; opacity: 1;"
            />
            <path
              class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
              cursor="unset"
              d="M0,-73.75L0,0Z"
              style="fill: yellow; opacity: 1;"
            />
            <path
              class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
              cursor="unset"
              d="M0,-73.75L0,0Z"
              style="fill: purple; opacity: 1;"
            />
            <path
              class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
              cursor="unset"
              d="M0,-73.75L0,0Z"
              style="fill: orange; opacity: 1;"
            />
          </g>
        </g>
        <g
          transform="translate(152.5, 200)"
        >
          <g>
            <text
              class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-c1eq84-MuiPieArcLabel-root"
              style="opacity: 0; transform: none;"
            />
            <text
              class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-c1eq84-MuiPieArcLabel-root"
              style="opacity: 0; transform: none;"
            />
            <text
              class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-c1eq84-MuiPieArcLabel-root"
              style="opacity: 0; transform: none;"
            />
            <text
              class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-c1eq84-MuiPieArcLabel-root"
              style="opacity: 0; transform: none;"
            />
            <text
              class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-c1eq84-MuiPieArcLabel-root"
              style="opacity: 0; transform: none;"
            />
            <text
              class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-c1eq84-MuiPieArcLabel-root"
              style="opacity: 0; transform: none;"
            />
          </g>
        </g>
      </g>
      <g
        class="MuiChartsLegend-root MuiChartsLegend-column css-1u0lry5-MuiChartsLegend-root"
      >
        <g
          class="MuiChartsLegend-series"
          transform="translate(355 125)"
        >
          <rect
            class="MuiChartsLegend-mark"
            fill="red"
            height="20"
            width="20"
            x="0"
            y="-10"
          />
          <text
            dominant-baseline="central"
            style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
            text-anchor="start"
            x="25"
            y="0"
          >
            <tspan
              dominant-baseline="central"
              dy="0px"
              x="25"
            >
              FOOD
            </tspan>
          </text>
        </g>
        <g
          class="MuiChartsLegend-series"
          transform="translate(355 155)"
        >
          <rect
            class="MuiChartsLegend-mark"
            fill="blue"
            height="20"
            width="20"
            x="0"
            y="-10"
          />
          <text
            dominant-baseline="central"
            style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
            text-anchor="start"
            x="25"
            y="0"
          >
            <tspan
              dominant-baseline="central"
              dy="0px"
              x="25"
            >
              TRANSPORTATION
            </tspan>
          </text>
        </g>
        <g
          class="MuiChartsLegend-series"
          transform="translate(355 185)"
        >
          <rect
            class="MuiChartsLegend-mark"
            fill="green"
            height="20"
            width="20"
            x="0"
            y="-10"
          />
          <text
            dominant-baseline="central"
            style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
            text-anchor="start"
            x="25"
            y="0"
          >
            <tspan
              dominant-baseline="central"
              dy="0px"
              x="25"
            >
              ENTERTAINMENT
            </tspan>
          </text>
        </g>
        <g
          class="MuiChartsLegend-series"
          transform="translate(355 215)"
        >
          <rect
            class="MuiChartsLegend-mark"
            fill="yellow"
            height="20"
            width="20"
            x="0"
            y="-10"
          />
          <text
            dominant-baseline="central"
            style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
            text-anchor="start"
            x="25"
            y="0"
          >
            <tspan
              dominant-baseline="central"
              dy="0px"
              x="25"
            >
              SERVICES
            </tspan>
          </text>
        </g>
        <g
          class="MuiChartsLegend-series"
          transform="translate(355 245)"
        >
          <rect
            class="MuiChartsLegend-mark"
            fill="purple"
            height="20"
            width="20"
            x="0"
            y="-10"
          />
          <text
            dominant-baseline="central"
            style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
            text-anchor="start"
            x="25"
            y="0"
          >
            <tspan
              dominant-baseline="central"
              dy="0px"
              x="25"
            >
              HEALTH
            </tspan>
          </text>
        </g>
        <g
          class="MuiChartsLegend-series"
          transform="translate(355 275)"
        >
          <rect
            class="MuiChartsLegend-mark"
            fill="orange"
            height="20"
            width="20"
            x="0"
            y="-10"
          />
          <text
            dominant-baseline="central"
            style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
            text-anchor="start"
            x="25"
            y="0"
          >
            <tspan
              dominant-baseline="central"
              dy="0px"
              x="25"
            >
              OTHER
            </tspan>
          </text>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
